config/kubernetes/distributed.yaml

```diff
@@ -0,0 +1,90 @@
+# Everything related to vector should be in this namespace
+apiVersion: v1
+kind: Namespace
+metadata:
+   name: logging
+   
+---
+# ConfigMap which contains vector.toml configuration for pods.
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vector-config
+  namespace: logging
+data:
+  vector-agent-config: |
+    # VECTOR.TOML
+    # Configuration for vector-agent
+
+    # Set global options
+    data_dir = "/var/lib/vector/"
+
+    # Ingest logs from Kubernetes
+    [sources.kubernetes_logs]
+      type = "kubernetes"
+
+    ######
+    # NOTE: Here, vector can be configured  
+    ######
+
+  # This line is not in VECTOR.TOML       
+
+---
+# Vector agent runned on each Node where it collects logs from pods.
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: vector-agent
+  namespace: logging
+spec:
+  minReadySeconds: 1
+  selector:
+    matchLabels:
+      name: vector-agent
+  template:
+    metadata:
+      labels:
+        name: vector-agent
+    spec:
+      volumes:
+      # Directory with logs
+      - name: var-log
+        hostPath:
+          path: /var/log/
+      # Docker log files in Kubernetes are symlinks to this folder.
+      #
+      # Vector can persist local data in here. Having that, updated
+      # Vector could continue where the last one stopped, if all the
+      # used sources have that capability.
+      - name: var-lib
+        hostPath:
+          path: /var/lib/
+      # Mount vector configuration from config map as a file vector.toml
+      - name: config-dir
+        configMap:
+         name: vector-config
+         items:
+           - key: vector-agent-config
+             path: vector.toml
+      containers:
+      - name: vector
+        image: timberio/vector:latest
+        imagePullPolicy: Always
+        args: ["-c","/etc/vector/vector.toml"]
+        volumeMounts:
+        - name: var-log
+          mountPath: /var/log/
+          readOnly: true        
+        - name: var-lib
+          mountPath: /var/lib
+        - name: config-dir
+          mountPath: /etc/vector
+          readOnly: true 
+      # /var/lib/vector/ needs to be created
```

Comment:
Is there a reason you don't just mount this dir as an emptyDir?

Reply:
Yes, more specifically the case of applying changed `distributed.yaml`. With host storage, checkpoints stored in that folder will be preserved between changes which will enable us to continue collecting logs where we stopped. With `emptyDir`, some logs will be lost in that case.  

Although there are some downsides. Major one is polluting the host.

But, now I see that there is an alternative. Vector is capable of updating it's topology with new configuration, although it will need to be triggered on file change. The question is, is it possible to make Kubernetes update already existing files created from `ConfigMap` when it changes. This would be a better solution, so I'll investigate.

Reply:
Hahaha, it does. I love Kubernetes. So it appears that yes, the file will change if `ConfigMap` is changed. 

I'll push a commit which uses emptyDir for storage, and open/push issue to listen for config file changes.

Reply:
Yes, unless the configmap is mounted with a subPath key

https://kubernetes.io/docs/concepts/storage/volumes/#configmap